cardano-config/src/Cardano/TracingOrphanInstances/Common.hs

```diff
@@ -70,3 +76,196 @@ deriving newtype instance FromJSON EpochNo
 instance ToObject Void where
   toObject _verb x = case x of {}
 
+-- Common failures
+
+instance ToObject (PredicateFailure (DELEG c)) where
+  toObject _verb (StakeKeyAlreadyRegisteredDELEG alreadyRegistered) =
+    mkObject [ "kind" .= String "StakeKeyAlreadyRegisteredDELEG"
+             , "Credential" .= String (textShow alreadyRegistered)
+             , "error" .= String "Staking credential already registered"
+             ]
+  toObject _verb (StakeKeyNotRegisteredDELEG notRegistered) =
+    mkObject [ "kind" .= String "StakeKeyNotRegisteredDELEG"
+             , "Credential" .= String (textShow notRegistered)
+             , "error" .= String "Staking credential not registered"
+             ]
+  toObject _verb (StakeKeyNonZeroAccountBalanceDELEG remBalance) =
+    mkObject [ "kind" .= String "StakeKeyNonZeroAccountBalanceDELEG"
+             , "Stake key remaining balance, if it exists" .= String (textShow remBalance)
+             ]
+  toObject _verb (StakeDelegationImpossibleDELEG unregistered) =
+    mkObject [ "kind" .= String "StakeDelegationImpossibleDELEG"
+             , "Unregistered credential" .= String (textShow unregistered)
```

Comment:
"Unregistered" is probably clear from the context, so this name could be shortened.